.github/workflows/appsignals-e2e-ec2-test.yml

```
@@ -58,6 +56,24 @@ jobs:
           role-to-assume: ${{ secrets.E2E_TEST_ROLE_ARN }}
           aws-region: ${{ env.AWS_DEFAULT_REGION }}
 
+      - uses: actions/download-artifact@v3
+        if: ${{ inputs.caller-workflow-name }} == 'main-build'
+        with:
+          name: aws-opentelemetry-agent.jar
+
+      - name: Upload main-build adot.jar to s3
+        if: ${{ inputs.caller-workflow-name }} == 'main-build'
+        run: aws s3 cp ./aws-opentelemetry-agent-*-SNAPSHOT.jar s3://main-build-adot-staging-jar/aws-opentelemetry-agent.jar
+
+      - name: Set Get ADOT.jar command environment variable
+        working-directory: testing/terraform/ec2
+        run: |
+          if [ ${{ inputs.caller-workflow-name }} == 'main-build' ]; then
+            echo APP_SIGNALS_GET_ADOT_JAR="aws s3 cp s3://main-build-adot-staging-jar/aws-opentelemetry-agent.jar ./adot.jar" >> $GITHUB_ENV
```

```suggestion
            echo TF_VAR_get_adot_jar="aws s3 cp s3://main-build-adot-staging-jar/aws-opentelemetry-agent.jar ./adot.jar" >> $GITHUB_ENV
```

You can simplify this code more by [using env vars](https://developer.hashicorp.com/terraform/language/values/variables#environment-variables). Terraform will pick this up automatically and you will not need to pass it in directly. This is up to you though, because using env vars to set TF variables can make it a bit more difficult to understand when and where the var is being set.